### PR TITLE
Front Page Announcements Element

### PR DESCRIPTION
**Pull request by mzagaja**
* Add carousel javascript to let user change the visible item on the home page on the front page
* Temporarily add announcements partial below the home page body
* Temporarily comment out .content-column to reflect new page style

Left Todo:
- [x] Back out old components from home page for final PR
- [x] Accessibility Check
- [x] Build out back end admin component
- [x] Add link button
- [x] If the uploaded image isn’t of a ratio/size that goes with the current component sizing then the image looks pretty odd. So we either need to make sure we just upload things with an appropriate ratio or I need to do some additional work around Dragonfly/Refinery and its use of thumbnails and auto-resizing.
- [x] Support 2x srcset images for retina browsers

### CODE REVIEW

**Comment by smithWEBtek** [comment]
@mzagaja I was able to create new announcements and verify that the javascript renders them as expected with clickable numbers (links).  
The images I uploaded with each announcement seem to have landed in a logical folder in `public/refinery/images ` but they are not rendering in the view.

**Comment by smithWEBtek** [comment]
@mzagaja Everything seems to work as expected. 

The only thing is image default sizing as you mentioned in comments above. 

I did notice that the position of the `LEARN MORE` button is dependent on the length of announcement body text, maybe it would be better in a fixed position lower left corner (?) or maybe its not an issue if the length of announcement body text is consistent.  

I tested one announcement with very long body text and it stretched out the image to fit the height of the huge text paragraph.

File: app/assets/stylesheets/partials/_announcements.scss [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        color: $color_font-light;
+    }
+
+    &__image {
+        color: $color_font-light;
+        flex-shrink: 2;
+        width: 66%;
+        max-width: 992px;
+        margin-bottom: 2rem;
+        object-fit: cover;
```
[the remark this answers is not in the document]
Comment (mzagaja) [reply app/assets/stylesheets/partials/_announcements.scss]: I will see if I can find an alternative

File: app/javascript/packs/carousel.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
+document.addEventListener("DOMContentLoaded", () => {
+  const carouselContainer = getCarousel();
+
+  if(carouselContainer) {
+    getData().then(result => populateCarousel(result, carouselContainer));
+  }
+});
+
+function getData() {
+  return new Promise((resolve, reject) => {
```
[the remark this answers is not in the document]
Comment (mzagaja) [reply app/javascript/packs/carousel.js]: Will look into this.

**Comment by mzagaja** [comment]
@ericyoungberg 

In regards to the rotation, that requirement was dropped from the issue per previous conversations with @theryankelly (and I just confirmed with him about that now) so we don't need to worry about the rotate/hover on this PR. We can add it in a future issue.

The javascript approach seemed to make the most sense to me given the original front end component I pull requested started from a JSON blob. I didn't consider doing it as an ERB because it seemed like doing work to override the pages view and adding components to it decorating the pages controller and relevant views would be a bit of a different rabbit hole and more tightly couple the refinery extensions. Though am open to refactoring if you see an easier path. Let me know.

Not sure if we would expect a screen reader to pick up content that's not visible to a user, but mostly didn't consider that due to lack of familiarity with the conventions around that. Open to more specific suggestions on improving this approach in that regard. Maybe rendering and hiding the elements and use CSS to swap them with `display: none`?

@smithWEBtek 

Yeah I'm not sure how much affordance we should give to longer body text. I think I'm ok with a moving button, but maybe an `overflow: scroll` could improve things here.

File: app/javascript/packs/carousel.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mzagaja) [reply app/javascript/packs/carousel.js]: Now working https://automate.browserstack.com/builds/51c97e649c34140b8e6966f6a5c53849892f827f/sessions/f2463b1070b94c904ae3a6d1ddab0df483b5a915

File: app/assets/stylesheets/partials/_announcements.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (mzagaja) [reply app/assets/stylesheets/partials/_announcements.scss]: Added a suggested polyfill from 
https://codepen.io/studiotwist/pen/XdgqmZ

**Pull request merged**